opendebates/tests/test_router.py

```diff
@@ -154,3 +154,43 @@ def test_readonly_three_replicas(self):
             # Round-robin should have come around by now
             self.assertEqual(out3, out0)
         self.assertFalse(was_db_written())
+
+
+class ContextManagerTest(TestCase):
+    def test_with_readwrite_db_when_readonly(self):
+        set_thread_readonly_db()
+        assert not is_thread_readwrite()
+        with readwrite_db():
+            assert is_thread_readwrite()
+        assert not is_thread_readwrite()
+
+    def test_with_readwrite_db_when_readwrite(self):
+        set_thread_readwrite_db()
+        assert is_thread_readwrite()
+        with readwrite_db():
+            assert is_thread_readwrite()
+        assert is_thread_readwrite()
+
+    def test_with_readonly_db_when_readonly(self):
+        set_thread_readonly_db()
+        assert not is_thread_readwrite()
+        with readonly_db():
+            assert not is_thread_readwrite()
+        assert not is_thread_readwrite()
+
+    def test_with_readonly_db_when_readwrite(self):
+        set_thread_readwrite_db()
+        assert is_thread_readwrite()
+        with readonly_db():
+            assert not is_thread_readwrite()
+        assert is_thread_readwrite()
+
+    def test_nesting(self):
```

Comment:
I guess it's more of a test of contextlib than our code, but it made me feel better.